Client/Frontend/Browser/Onboarding/OnboardingRewardsAgreementViewController.swift

```diff
@@ -20,27 +47,335 @@ class OnboardingRewardsAgreementViewController: OnboardingViewController {
+        view.backgroundColor = #colorLiteral(red: 0.1176470588, green: 0.1254901961, blue: 0.1607843137, alpha: 1)
+        
+        contentView.turnOnButton.addTarget(self, action: #selector(turnOnTapped), for: .touchUpInside)
         contentView.skipButton.addTarget(self, action: #selector(skipTapped), for: .touchUpInside)
         
-        (view as! View).onTermsOfServicePressed = { [weak self] in  // swiftlint:disable:this force_cast
-            guard let self = self,
-                  let profile = self.profile else { return }
-            
-            self.present(OnboardingWebViewController(profile: profile), animated: true, completion: nil)
+        contentView.onTermsOfServicePressed = { [weak self] in
+            guard let self = self else { return }
+            self.present(OnboardingWebViewController(profile: self.profile, url: .termsOfService), animated: true, completion: nil)
+        }
+        
+        contentView.onPrivacyPolicyPressed = { [weak self] in
+            guard let self = self else { return }
+            self.present(OnboardingWebViewController(profile: self.profile, url: .privacyPolicy), animated: true, completion: nil)
         }
     }
     
-    override func continueTapped() {
+    @objc
+    private func skipTapped() {
+        onOnboardingStateChanged?(self, .skipped)
+    }
+    
+    @objc
+    private func turnOnTapped() {
         Preferences.General.basicOnboardingProgress.value = OnboardingProgress.rewards.rawValue
+        
+        rewards.isEnabled = true
+        onOnboardingStateChanged?(self, .complete)
+    }
+}
+
+extension OnboardingRewardsAgreementViewController {
+    private struct UX {
+        /// The onboarding screens are showing as a modal on iPads.
+        static let preferredModalSize = CGSize(width: 375, height: 667)
+        
+        /// A negative spacing is needed to make rounded corners for details view visible.
+        static let negativeSpacing = -16.0
+        static let descriptionContentInset = 25.0
+        static let linkColor: UIColor = .braveBlurple
+        static let animationContentInset = 50.0
+        static let checkboxInsets = -44.0
+    }
+    
+    private struct Views {
+        static func primaryButton(text: String = Strings.OBContinueButton) -> UIButton {
+            let button = RoundInterfaceButton().then {
+                $0.setTitle(text, for: .normal)
+                $0.backgroundColor = .braveOrange
+                $0.contentEdgeInsets = UIEdgeInsets(top: 12, left: 25, bottom: 12, right: 25)
+            }
             
-        super.continueTapped()
+            return button
+        }
+        
+        static func secondaryButton(text: String = Strings.OBSkipButton) -> UIButton {
+            let button = UIButton().then {
+                $0.setTitle(text, for: .normal)
+                let titleColor = #colorLiteral(red: 0.5176470588, green: 0.5333333333, blue: 0.6117647059, alpha: 1)
```

Comment:
we can see if we have a predefined color for it already in `DesignSystemColor`

Reply:
This is a color we dont haveut I added this color to UX struct and used rgb initializer  https://github.com/brave/brave-ios/pull/4508/commits/858063434f530cde5c6e9ff8394401225f85ef09